### PR TITLE
CI: integration tests: use runner as py module

### PR DESCRIPTION
**Pull request by maxknv**
#no_merge_commit
 #ci_set_integration
 #batch_0

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

1.  run ci_runner as a py module (not separate process) to ease debugging

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 9dd3ae05f86b408035cf924caee32abe57a1948a with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61658/9dd3ae05f86b408035cf924caee32abe57a1948a/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: tests/ci/ci_runner.py [hunk 1]
```diff
@@ -102,7 +104,7 @@ def get_counters(fname):
         "PASSED": set([]),
         "FAILED": set([]),
         "SKIPPED": set([]),
-    }
+    }  # type: Dict[str, Any]
```
Comment (Felixoid) [inline tests/ci/ci_runner.py:107]: `Set[str]`

File: tests/ci/ci_runner.py [hunk 2]
```diff
@@ -292,7 +294,7 @@ def get_image_with_version(self, name):
             return name + ":latest"
         return name
 
-    def get_image_version(self, name: str):
+    def get_image_version(self, name: str) -> Any:
```
Comment (Felixoid) [inline tests/ci/ci_runner.py:297]: str?

File: tests/ci/ci_runner.py [hunk 3]
```diff
@@ -380,15 +382,15 @@ def _install_clickhouse(self, debs_path):
         os.chmod(CLICKHOUSE_ODBC_BRIDGE_BINARY_PATH, 0o777)
         os.chmod(CLICKHOUSE_LIBRARY_BRIDGE_BINARY_PATH, 0o777)
         shutil.copy(
-            CLICKHOUSE_BINARY_PATH, os.getenv("CLICKHOUSE_TESTS_SERVER_BIN_PATH")
+            CLICKHOUSE_BINARY_PATH, os.getenv("CLICKHOUSE_TESTS_SERVER_BIN_PATH")  # type: ignore
```
Comment (Felixoid) [inline tests/ci/ci_runner.py:385]: should be `str`, why mypy complains?

**Review by Felixoid — COMMENTED** [review]
I'd call `tests/ci/integration_runner.py`. Besides that, small notes about the typing.

Looks very straight forward, let's ask @qoega as well

File: tests/ci/ci_runner.py (hunk 3 above)
Comment (maxknv) [reply tests/ci/ci_runner.py:385]: might be None

Comment (Felixoid) [reply tests/ci/ci_runner.py:385]: If it's `None`, the `shutil.copy` must fail. So, it is worth setting a decent default or failing explicitly.

File: tests/ci/ci_runner.py (hunk 1 above)
Comment (maxknv) [reply tests/ci/ci_runner.py]: I put my best reasonable efforts to pass the style_check

**Pull request merged**